test/support/tdb_catch.h

```diff
@@ -35,6 +35,36 @@
 #ifndef TILEDB_MISC_TDB_CATCH_H
 #define TILEDB_MISC_TDB_CATCH_H
 
+/*
+ * The header-only version of Catch includes Windows system headers that bleed
+ * out preprocessor definitions. This has been reported as
+ * https://github.com/catchorg/Catch2/issues/2432. It may not be a problem for
+ * version 3 of Catch, which isn't header-only.
+ *
+ * We need to detect and remove definitions that are a problem. It shouldn't be
+ * strictly necessary, but out of a superabundance of caution, we detect
+ * previous definitions and leave them unchanged.
+ *   - DELETE
+ */
+#if defined(DELETE)
+#define TILEDB_CATCH_DELETE_PREDEFINED DELETE
+#endif
+
+/*
+ * The actual payload of this file
+ */
 #include <catch.hpp>
 
+/*
+ * Clean up preprocessor definitions
+ */
+#if defined(TILEDB_CATCH_DELETE_PREDEFINED)
+#define DELETE TILEDB_CATCH_DELETE_PREDEFINED
```

Comment:
This approach I think is still going to lead to failure to compile with the new unqualified DELETE enum ... DELETE will still be left with definition that is at worst might -silently-but-incorrectly-compile- and at best still fail to compile.  (That's why you're having to '#undef DELETE' whatever 'DELETE' happens to be at this point.

Reply:
Appears to be compiling, @eric-hughes-tiledb can this be resolved?

Reply:
It's resolved. #3504 is using this code already, and @dhoke4tdb addressed his concern about an undetected problem at the point of symptom by adding detection for it.